src/main/webapp/app/exercises/shared/import/exercise-import.component.ts

```diff
@@ -42,9 +48,32 @@ export class TextExerciseImportComponent implements OnInit {
     isCourseFilter = true;
     isExamFilter = true;
 
-    constructor(private pagingService: TextExercisePagingService, private sortService: SortService, private activeModal: NgbActiveModal) {}
+    titleKey: string;
+
+    constructor(private sortService: SortService, private activeModal: NgbActiveModal, private injector: Injector) {}
 
     ngOnInit(): void {
+        if (!this.exerciseType || this.exerciseType == ExerciseType.FILE_UPLOAD) {
+            return;
+        }
+        switch (this.exerciseType) {
```

Comment:
I would move this into an own function `getPagingService(exerciseType, injector) -> PagingService` so that we don't need those break statements and can directly return